### PR TITLE
feat: cpp extension for LPC

### PR DESCRIPTION
**Pull request by yoyolicoris**
C++ kernel for LPC adapted from #10. Will close #10 after merged.
Compiled with OpenMP if available.
The following is the benchmark (`order=20`) compared to v0.6.
```
[--------------------------  --------------------------]
                                |  numba_lpc  |  lpc_cpu
4 threads: ---------------------------------------------
      bs_1__n_16384__threads_4  |     341.0   |    307.1
      bs_2__n_16384__threads_4  |     343.5   |    392.9
      bs_4__n_16384__threads_4  |     374.7   |    423.4
      bs_8__n_16384__threads_4  |    2434.8   |   3099.4

Times are in microseconds (us).
```

The inner for-loop is not parallelised, though.
I tried some SIMD instructions with OpenMP but didn't succeed.
It should be fine since the runtime is just slightly slower than Numba.

### CODE REVIEW

File: torchlpc/core.py [hunk 1]
```diff
@@ -159,7 +161,12 @@ class LPC(Function):
     def forward(x: torch.Tensor, A: torch.Tensor, zi: torch.Tensor) -> torch.Tensor:
         if x.is_cuda:
             y = lpc_cuda(x.detach(), A.detach(), zi.detach())
+        elif EXTENSION_LOADED:
+            y = torch.ops.torchlpc.lpc_cpu(x, A, zi)
```
Comment (christhetree) [inline torchlpc/core.py:165]: Just curious, why do you call `.detach()` for GPU but not for CPU?

Comment (yoyolicoris) [reply torchlpc/core.py:165]: To properly expose the underneath storage to the other framework, at least this is the case for numba. For Torch C++ API, this is not necessary, as they accept Tensor type.

File: tests/test_extension.py [hunk 2]
```diff
@@ -33,3 +33,22 @@ def test_scan_cpu_equiv(samples: int, cmplx: bool):
...
+@pytest.mark.parametrize(
+    "samples",
+    [1024],
+)
+@pytest.mark.parametrize(
+    "cmplx",
+    [True, False],
+)
+def test_lpc_cpu_equiv(samples: int, cmplx: bool):
+    batch_size = 4
```
Comment (christhetree) [inline tests/test_extension.py:47]: You don't need to change this now, but in general I think it's worth parameterizing `batch_size` as well and testing batch_size = 1 and batch_size > 1 since this exposes `.squeeze()` and related errors.

**Review by christhetree — APPROVED** [review]
Sorry for the delay, left two comments, but the changes look good. Feel free to close my PR if you like. Maybe `benchmark_forward.py` could be useful from that PR, but otherwise this covers everything.

File: tests/test_extension.py (hunk 2 above)
Comment (yoyolicoris) [reply tests/test_extension.py:47]: Fair. Will add it in the future.
